### DISCUSSION TITLE
Hide elements for logged on user...

### DISCUSSION THREAD

**Discussion by JulianKirkness** [cortexrd/Knack-Toolkit-Library#400] [General]
I have a client who's Knack app is used for a range of HR processes (amongst other things) and they would like to be able to stop people from accessing a page (or specific views) for their own records (account records are the basic staff record also).

Ideally this would mean being able to hide a link to a page in a grid view row for the logged in user - or to hide views on forms for the logged in user.

Is there any way to do this with the KTL? (Or any other way)

**Comment by CSWinnall** [comment]
Hi Julian

I am not sure this can be done for links in a grid by KTL.

For links in a grid you can use a combination of code and KTL.
You will need to add the column for the created by field then you can use `_hc=columnHeader` to hide it.

Then add this function to your JS pane:
```
function getTableRows(viewId, callback) {
    const viewElement = $(`#${viewId}`);
    const tableRows = viewElement.find('table tbody tr').not('.kn-table-group');
    tableRows.each((index, row) => {
        callback(index, $(row));
    });
}
```
Then add this code, replace the field ID (Record Creator) and the view ID with yours and also replace Column Header with the header of the account field you are hiding:

```
$(document).on('knack-view-render.view_xyz', function (_, {key: viewId}) {
    const loggedInUserId = Knack.getUserAttributes().id;
    getTableRows(viewId, (i, row) => {
        const createdById = row.find('td.field_abc span span').attr('class');
        if (createdById === loggedInUserId) {
            const columnIndex = ktl.views.getColumnIndex(viewId, 'Column Header')
            if (columnIndex) row.find(`td:nth-child(${columnIndex}) a`).remove();
        }
    });
});
```
Let me know if this works or if you need any more help.

Craig
